src/SleighInstruction.h

```diff
@@ -0,0 +1,452 @@
+/* radare - LGPL - Copyright 2020 - FXTi */
+
+#ifndef R2GHIDRA_SLEIGHINSTRUCTION_H
+#define R2GHIDRA_SLEIGHINSTRUCTION_H
+
+#include <vector>
+#include <unordered_set>
+#include "architecture.hh"
+#include "sleigh_arch.hh"
+#include "crc32.hh"
+#include <unordered_map>
+
+/**
+ * There is still room for optimization, Now SleighInstruction
+ * is actually SleighInstructionPrototype in JAVA codebase.
+ * But changes to per-instruction implementation.
+ * In JAVA codebase, SleighInstructionPrototype is Constructor
+ * pattern of one kind of instructions. And concrete flow destination
+ * address will be resolved on actual buffer of specified instruction.
+ * That's why JAVA codebase cache SleighInstructionPrototype in SleighLanguage.
+ * Multiple InstructionContext will be mapped to one single cache prototype,
+ * which will save time and space of collecting Constructors.
+ * But to implement that in C++ codebase, you will have to create something
+ * like InstructionContext and move necessary status and API to that.
+ */
+
+class R2Sleigh;
+class R2DisassemblyCache;
+class SubParserWalker;
+class SleighParserContext;
+class ExportHelper
+{
+	// Please keep all structures in this class sync with origin!!!
```

Comment:
These structures have to be removed before merge. You can send a pr to https://github.com/thestr4ng3r/ghidra to expose things.